src/lsm/scan_tree.zig

```diff
@@ -821,6 +763,96 @@ fn ScanTreeLevelType(comptime ScanTree: type, comptime Storage: type) type {
             }
         }
 
+        /// Move to the next `value_block` or `table_info` according to the current state.
+        fn move_next(self: *ScanTreeLevel) void {
+            assert(self.values == .fetching or
+                self.values == .buffered);
+
+            switch (self.state) {
+                .iterating_manifest => self.move_manifest_table_next(null),
+                .loading_index => unreachable,
+                .iterating_values => |*iterating_values| {
+                    if (iterating_values.iterator.empty()) {
+                        switch (iterating_values.manifest_table_next) {
+                            .key_exclusive => |key_exclusive| {
+                                assert(switch (self.scan.direction) {
+                                    .ascending => self.scan.key_min <= key_exclusive,
+                                    .descending => self.scan.key_max >= key_exclusive,
+                                });
+                                // Load the next `table_info`.
+                                self.state = .iterating_manifest;
+                                self.values = .fetching;
+                                self.move_manifest_table_next(key_exclusive);
+                            },
+                            .out_of_range => {
+                                // The next `table_info` is out of the key range, so it's finished.
+                                self.state = .{ .finished = .{} };
+                                self.values = .finished;
+                            },
+                        }
+                    } else {
+                        // Keep iterating to the next `value_block`.
+                        self.values = .fetching;
+                    }
+                },
+                .finished => unreachable,
+            }
+        }
+
+        /// Moves the iterator to the next `table_info` that might contain the key range.
+        fn move_manifest_table_next(
```

Comment:
I'd call it `move_next_manifest_table`: the base here is `move_next`, within this larger `move_next` we have `move_next_manifest_table` suboperation. In this case, the qualifier is not "next", but rather "manifest table".

A general rule of thumb: if you have function `foo` and it calls function `bar`, and no-one else calls `bar`, then `bar` probably should be called `foo_bar` --- _likely_ it'll follow MSB semantics, _and_ it'll sort nicely in the source file!